hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/ExponentialRetryPolicy.java

```diff
@@ -75,23 +57,16 @@ public class ExponentialRetryPolicy extends AbfsRetryPolicy {
    */
   private final int minBackoff;
 
-  /**
-   * Initializes a new instance of the {@link ExponentialRetryPolicy} class.
-   */
-  public ExponentialRetryPolicy(final int maxIoRetries) {
-
-    this(maxIoRetries, DEFAULT_MIN_BACKOFF, DEFAULT_MAX_BACKOFF,
-        DEFAULT_CLIENT_BACKOFF);
-  }
 
   /**
    * Initializes a new instance of the {@link ExponentialRetryPolicy} class.
    *
-   * @param conf The {@link AbfsConfiguration} from which to retrieve retry configuration.
+   * @param conf         The {@link AbfsConfiguration} from which to retrieve retry configuration.
+   * @param maxIoRetries The maximum number of retry attempts
    */
-  public ExponentialRetryPolicy(AbfsConfiguration conf) {
-    this(conf.getMaxIoRetries(), conf.getMinBackoffIntervalMilliseconds(), conf.getMaxBackoffIntervalMilliseconds(),
-        conf.getBackoffIntervalMilliseconds());
+  public ExponentialRetryPolicy(AbfsConfiguration conf, Integer maxIoRetries) {
```

Comment:
This should not have maxIoRetries as a parameter as that we can derive from the abfsConfiguration